app/assets/javascripts/common/reports.js

```diff
@@ -1253,9 +1394,70 @@ function withBaseBarConfig(config) {
   );
 }
 
+function mergeConfig(baseConfig, overwritingConfig) {
+  return _.mergeWith(
+    baseConfig,
+    overwritingConfig,
+    mergeArraysWithConcatenation
+  );
+}
 
 function mergeArraysWithConcatenation(objValue, srcValue) {
   if (_.isArray(objValue)) {
     return objValue.concat(srcValue);
   }
 }
+
+function isGoalLineEnabled(domElement) {
+  const cardNodeBPControlled = domElement;
+  const isEnabled = cardNodeBPControlled.querySelector("[data-show-goal-lines]")
+    .dataset.showGoalLines;
+  if (isEnabled === "true") {
+    return true;
+  }
+  return false;
+}
+
+function addGoalLineToConfig(config, periodValues, goalDownwards = false) {
+  const latestDecValue = getLatestDecemberValue(periodValues);
+  const goal = calculateGoal(latestDecValue, goalDownwards);
+  const goalLineConfig = {
+    plugins: [goalLinePlugin(goal)],
+  };
+  return mergeConfig(config, goalLineConfig);
+}
+
+function calculateGoal(value, goalDownwards) {
+  if (goalDownwards) {
+    return calculateGoalDownwards(value);
+  }
+  return calculateGoalUpwards(value);
+}
+
+function calculateGoalUpwards(decemeberValue) {
```

Comment:
[nit] Typo in `decemeberValue`.

Reply:
Moving this calculation to the backend (to the controller) and keeping just the presentation logic here would be easier to maintain. We wouldn't have to look for the last `Dec` value in the data which couples this to the date format.

I think we can ship this and create a story for the server-devs to move it to the rails controller when they pick this up.

Reply:
Fixed 'decemberValue'

Reply:
Created story -> https://app.shortcut.com/simpledotorg/story/10512/maintenance-move-calculategoalupwards-function-into-ruby-and-pass-value